lisa/operating_system.py

```diff
@@ -443,7 +464,7 @@ def get_package_information(
     def get_repositories(self) -> List[RepositoryInfo]:
         raise NotImplementedError("get_repositories is not implemented")
 
-    def add_azure_core_repo(self) -> None:
+    def add_azure_core_repo(self, repo_name: Optional[AzureCoreRepo] = None) -> None:
```

Comment:
Should the repo name is set by distro class? It doesn't need to be passed in from caller.

Reply:
This **repo_name** is primarily used for Debian images. In the past, only tvm test cases required the addition of the **azurecore** repo. The test case **verify_measuredboot_compatibility** needs the **azure-compatscanner** package, which is absent in the **azurecore-debian** repo, so the **azurecore** repo from Ubuntu Bionic is used instead.

However, if more test cases require the **azurecore** repo, using Ubuntu's repo for Debian images would result in the **azurecore-debian** repo not being covered in tests.


To address this, a change has been made. By default, if the **repo_name** parameter is None, Debian Buster images will use the **azurecore-debian** repo. In other scenarios, such as when the "**azure-compatscanner**" package is required, the caller can define the use of the **azurecore** repo from Ubuntu.

Reply:
Ok. If it's for increasing coverage, it looks good to me. If it has to be specified for different distro, it should be handled inside of the os classes.